File: Projekt1_cpp/Projekt1_cpp/Zolw.cpp
#include "Zolw.h"

Zolw::Zolw(pair<int, int>pole, Swiat* swiat)
	:Zwierze(2, 1, 0, pole, swiat) {}

Organizm* Zolw::potomek(pair<int, int>pole) {
	return new Zolw(pole, swiat);
}

void Zolw::akcja() {
	if (czyRusza() == true) {
		pair<int, int> nowePole = losujPole(false);
		if (swiat->getSwiat()[nowePole.first][nowePole.second]->getNazwa() == "puste") {
			ruch(nowePole);
		}
		else
			kolizja(swiat->getSwiat()[nowePole.first][nowePole.second]);
	}
}

void Zolw::rysowanie() {
	cout << 'z';
}

string Zolw::rodzajKolizji(Organizm* organizm) {
	if (organizm->getSila() < 5)
		return "Odepchniety";
	else if (organizm->getSila() >= sila)
		return "Wygrywa";
	else
		return "Przegrywa";
}

string Zolw::getNazwa() {
	return "zolw";
}

bool Zolw::czyRusza() {
	int los = rand() % 100;
	if (los <= 75)
		return true;
	return false;
}

File: Projekt2_java/src/projekt2/zwierzeta/Zolw.java
package projekt2.zwierzeta;

import projekt2.*;

import java.awt.*;

public class Zolw extends Zwierze {

    public Zolw(Pair pole, Swiat swiat) {
        super(2, 1, 0, pole, swiat);
    }

    @Override
    public Organizm potomek(Pair pole) {
        return new Zolw(pole, swiat);
    }

    @Override
    public void akcja() {
        if (czyRusza() == true) {
            super.akcja();
        }
    }

    @Override
    public Color rysuj() {
        return Color.GREEN.darker();
    }

    @Override
    public String rodzajKolizji(Organizm organizm) {
        if (organizm.getSila() < 5)
            return "Odepchniety";
        return super.rodzajKolizji(organizm);
    }

    @Override
    public String getNazwa() {
        return "zolw";
    }

    private boolean czyRusza() {
        int los = swiat.rand.nextInt(100);
        if (los <= 75)
            return true;
        return false;
    }
}
